ui/server/src/main/scala/pl/touk/nussknacker/ui/definition/validator/ParameterValidatorsExtractorChain.scala

```diff
@@ -0,0 +1,25 @@
+package pl.touk.nussknacker.ui.definition.validator
+
+import pl.touk.nussknacker.engine.api.definition.{Parameter, ParameterValidator}
+import pl.touk.nussknacker.engine.api.process.ParameterConfig
+
+object ParameterValidatorsExtractorChain {
+
+  def apply(parameterConfig: ParameterConfig): ParameterValidatorsExtractorChain = {
+    val strategies = Seq(
+      new ParameterConfigValidatorsExtractor(parameterConfig),
+      new ParameterValidatorsExtractor
+    )
+    new ParameterValidatorsExtractorChain(strategies)
+  }
+}
+
+class ParameterValidatorsExtractorChain(strategies: Iterable[ParameterValidatorsExtractorStrategy]) {
+
+  def evaluate(param: Parameter): List[ParameterValidator] = {
```

Comment:
why we cannot add some validators via config and leave rest from methods?

Reply:
I we would merge validators from config with those from annotations then you would not be able to define in config that you want f.e. empty list of validators instead of default ones

Reply:
One option that I see is that we can sometimes want to remove some validator. Currently all non optional, not nullable parameters will have mandatory validator. If will be only possibility to add new one, you won't be able to remove this validator without adding `@Nullable` annotation...

Reply:
Are we agree with that?

Reply:
Yes, I see your point. This indeed enables some cases, but OTOH it's slightly non-intuitive... Maybe at least let's add comment somewhere?

Reply:
Added above apply method